frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -0,0 +1,186 @@
+    <DataListItemRow>
+      <DataListCheck
+        isDisabled={(disableUncheckedRow && !column.visible) || READ_ONLY_COLS.includes(column.ID)}
+        aria-labelledby={`table-column-management-item-${column.ID}`}
+        checked={column.visible}
+        name={column.title}
+        id={column.ID}
+        onChange={onChange}
+      />
+      <DataListItemCells
+        dataListCells={[
+          <DataListCell id={`table-column-management-item-${column.ID}`} key={column.ID}>
+            {column.title}
+          </DataListCell>,
+        ]}
+      />
+    </DataListItemRow>
+  </DataListItem>
+);
+
+export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
+  cancel,
+  close,
+  columnManagementID,
+  columnManagementType,
+  selectedColumns,
+}) => {
+  const dispatch = useDispatch();
+  const initialDefaultColumns = !_.isEmpty(selectedColumns)
+    ? selectedColumns.filter((column) => column.title.length > 0 && !column.additional)
+    : {};
+  const initialAdditionalColumns = !_.isEmpty(selectedColumns)
+    ? selectedColumns.filter((column) => column.additional)
+    : {};
+  const [defaultColumns, setDefaultColumns] = React.useState<any[]>(initialDefaultColumns);
+
+  const [additionalColumns, setAdditionalColumns] = React.useState<ColumnHeaderRow[]>(
+    initialAdditionalColumns,
+  );
+
+  const updateColumns = (checked: boolean, id: string, columnArray: any): any => {
+    return columnArray.map((column) => {
+      if (!id || column.ID === id) {
+        return { ...column, visible: checked };
+      }
+      return column;
+    });
+  };
+
+  const defaultColumnChange = (checked: boolean, event: React.SyntheticEvent): void => {
+    setDefaultColumns(updateColumns(checked, event?.currentTarget?.id, defaultColumns));
+  };
+
+  const additionalColumnChange = (checked: boolean, event: React.SyntheticEvent): void => {
+    setAdditionalColumns(updateColumns(checked, event?.currentTarget?.id, additionalColumns));
+  };
+
+  const submit = (event): void => {
+    event.preventDefault();
+    dispatch(setTableColumns(columnManagementID, [...defaultColumns, ...additionalColumns]));
+    close();
+  };
+
+  const areMaxColumnsDisplayed =
+    additionalColumns.filter((column) => column.visible).length +
+      defaultColumns.filter((column) => column.visible).length >=
+    MAX_VIEW_COLS;
+
+  const resetColumns = (event: React.SyntheticEvent): void => {
+    event.preventDefault();
+    setDefaultColumns(updateColumns(true, null, defaultColumns));
+    setAdditionalColumns(updateColumns(false, null, additionalColumns));
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle className="modal-header">Manage Columns</ModalTitle>
+      <ModalBody>
+        <div className="co-m-form-row">
+          <p>Checked categories will be displayed in the table.</p>
+        </div>
+        <div className="row co-m-form-row">
```

Comment:
Is `row` needed here?

Reply:
changes spacing a bit but not needed here. more in line with the design without it...removed. thanks